editor/src/components/navigator/left-pane/pages-pane.tsx

```diff
@@ -282,12 +321,146 @@ const PageRouteEntry = React.memo<PageRouteEntryProps>((props) => {
           },
         }}
       >
-        {lastTemplateSegment}
+        {lastResolvedSegment}
+      </span>
+    </FlexRow>
+  )
+})
+
+interface FavoriteEntryProps {
+  favorite: string
+  active: boolean
+  addedToFavorites: boolean
+}
+
+const FavoriteEntry = React.memo(({ favorite, active, addedToFavorites }: FavoriteEntryProps) => {
+  const [navigationControls] = useAtom(RemixNavigationAtom)
+  const [activeRemixScene] = useAtom(ActiveRemixSceneAtom)
+
+  const onClick = React.useCallback(() => {
+    void navigationControls[EP.toString(activeRemixScene)]?.navigate(favorite)
+  }, [navigationControls, activeRemixScene, favorite])
+
+  return (
+    <FlexRow
+      style={{
+        flexShrink: 0,
+        color: active ? colorTheme.neutralForeground.value : colorTheme.subduedForeground.value,
+        backgroundColor: active ? colorTheme.subtleBackground.value : 'transparent',
+        border: addedToFavorites
+          ? '1px solid transparent'
+          : '1px dashed ' + colorTheme.border3.value,
+        boxSizing: 'border-box',
+        marginLeft: 8,
+        marginRight: 8,
+        paddingLeft: 19, // to visually align the icons with the route entries underneath the favorites section
+        paddingTop: 3,
+        paddingBottom: 3,
+        height: UtopiaTheme.layout.rowHeight.smaller,
+        alignItems: 'center',
+        borderRadius: 2,
+        position: 'relative',
+      }}
+      onClick={onClick}
+    >
+      <Icn
+        style={{
+          marginRight: 0,
+        }}
+        category='filetype'
+        color={active ? 'main' : 'secondary'}
+        type={'other'}
+        width={12}
+        height={12}
+      />
+      {/* TODO if we want renaming, cannibalize it from FileBrowserItem */}
+      <span
+        style={{
+          marginLeft: 6,
+          display: 'inline-block',
+          whiteSpace: 'nowrap',
+          overflow: 'hidden',
+          textOverflow: 'ellipsis',
+          flexGrow: 1,
+        }}
+      >
+        {favorite}
       </span>
+      <StarUnstarIcon url={favorite} selected={active} addedToFavorites={addedToFavorites} />
     </FlexRow>
   )
 })
 
+interface StarUnstarIconProps {
+  url: string
+  addedToFavorites: boolean
+  selected: boolean
+}
+
+const StarUnstarIcon = React.memo(({ url, addedToFavorites, selected }: StarUnstarIconProps) => {
+  const dispatch = useDispatch()
+
+  const onClickAddOrRemoveFavorite = React.useCallback(
+    (e: React.MouseEvent) => {
+      if (!addedToFavorites) {
+        dispatch([addNewFeaturedRoute(url)])
+      } else {
+        dispatch([removeFeaturedRoute(url)])
+      }
+      e.stopPropagation()
+    },
+    [dispatch, url, addedToFavorites],
+  )
+
+  const [mouseOver, setMouseOver] = React.useState(false)
```

Comment:
I tried to not rely on mouseOver events,and instead have two Icn components below each other, but I gave up after half an hour of struggling.